tests/std/tests/VSO_0000000_regex_use/test.cpp

```diff
@@ -1698,6 +1698,190 @@ void test_gh_5377() {
+        test_regex quantified_empty_regex_bre(&g_regexTester, R"(\(\)*)", option);
+        quantified_empty_regex_bre.should_search_match_capture_groups("", "", match_default, {{0, 0}});
+        quantified_empty_regex_bre.should_search_match_capture_groups("b", "", match_default, {{0, 0}});
+    }
+
+
+    for (auto option : {ECMAScript, extended, egrep, awk}) {
+        test_regex simple_sequence_regex_ecma_or_ere(&g_regexTester, "(ab)*", option);
+        simple_sequence_regex_ecma_or_ere.should_search_match_capture_groups("", "", match_default, {{-1, -1}});
+        simple_sequence_regex_ecma_or_ere.should_search_match_capture_groups("b", "", match_default, {{-1, -1}});
+        simple_sequence_regex_ecma_or_ere.should_search_match_capture_groups("ababcc", "abab", match_default, {{2, 4}});
+    }
+
+    for (auto option : {basic, grep}) {
+        test_regex simple_sequence_regex_bre(&g_regexTester, R"(\(ab\)*)", option);
+        simple_sequence_regex_bre.should_search_match_capture_groups("", "", match_default, {{-1, -1}});
+        simple_sequence_regex_bre.should_search_match_capture_groups("b", "", match_default, {{-1, -1}});
+        simple_sequence_regex_bre.should_search_match_capture_groups("ababcc", "abab", match_default, {{2, 4}});
+    }
+
+    for (string pattern : {"(ab(?=ab))*", "(ab(?!cc))*"}) {
+        test_regex nested_assertion_regex_ecma(&g_regexTester, pattern, ECMAScript);
+        nested_assertion_regex_ecma.should_search_match_capture_groups("", "", match_default, {{-1, -1}});
+        nested_assertion_regex_ecma.should_search_match_capture_groups("b", "", match_default, {{-1, -1}});
+        nested_assertion_regex_ecma.should_search_match_capture_groups("ababcc", "ab", match_default, {{0, 2}});
+        nested_assertion_regex_ecma.should_search_match_capture_groups("abababcc", "abab", match_default, {{2, 4}});
+    }
+
+    for (string pattern : {"(a*)*", "(a?)*", "(a?)?"}) {
+        test_regex nested_quantifier_regex_ecma(&g_regexTester, pattern, ECMAScript);
+        nested_quantifier_regex_ecma.should_search_match_capture_groups("", "", match_default, {{-1, -1}});
+        nested_quantifier_regex_ecma.should_search_match_capture_groups("b", "", match_default, {{-1, -1}});
+        nested_quantifier_regex_ecma.should_search_match_capture_groups("a", "a", match_default, {{0, 1}});
+
+        for (auto option : {extended, egrep, awk}) {
+            test_regex nested_quantifier_regex_ere(&g_regexTester, pattern, option);
+            nested_quantifier_regex_ere.should_search_match_capture_groups("", "", match_default, {{0, 0}});
+            nested_quantifier_regex_ere.should_search_match_capture_groups("b", "", match_default, {{0, 0}});
+            nested_quantifier_regex_ere.should_search_match_capture_groups("a", "a", match_default, {{0, 1}});
+        }
+    }
+
+    for (auto option : {basic, grep}) {
+        test_regex doublestar_regex_bre(&g_regexTester, R"(\(a*\)*)", option);

```

Comment:
Similarly, we could translate the other `nested_quantifier` test cases to BRE, which would also address my nitpick that `doublestar` inconsistently lacks an underscore to separate the words.